lib/decorators/actions.js

```diff
@@ -101,8 +101,16 @@ export function unfocus({ commit, dispatch, state }) {
   const current = _.get(state, 'ui.currentFocus');
 
   if (current) {
-    commit(UN_FOCUS);
-    return dispatch('closeForm');
+    const formSchema = state.schemas[getComponentName(current.uri)],
```

Comment:
I think we can get rid of some conditions here

```js
if (current) {
    const formSchema = state.schemas[getComponentName(current.uri)],
      allowUnfocus = typeof formSchema.formValidation === 'function' ? formSchema.formValidation() : true;


    if (allowUnfocus) {
      commit(UN_FOCUS);
      
      return dispatch('closeForm');
    }
  } 
  
  return Promise.resolve();
```

Reply:
👍